### PR TITLE
Add new Task documentation: Telepresence

### PR DESCRIPTION
**Pull request by richarddli**
Telepresence is a popular tool for local development & debugging against a remote Kubernetes cluster.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4955)

### CODE REVIEW

**Comment by richarddli** [comment]
Thanks! I've made the changes.

File: docs/tasks/debug-application-cluster/local-debugging.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Getting a shell on a remote cluster
+
+If you start `telepresence` with no arguments, you'll get a special shell. Unlike a remote shell, the telepresence shell runs locally, giving you full access to your filesystem.
+
+There are a variety of ways that the local shell can be used. For example, you could write a shell script on your laptop, and run it directly from the shell in real-time. You can do this on a remote shell as well, of course, but you may not be able to use your preferred code editor, and the script is deleted when you terminate the container.
```
Comment (radhikapc) [inline docs/tasks/debug-application-cluster/local-debugging.md]: The `telepresence` shell can be used in a variety of ways. For example, write a shell script on your laptop, and run it directly from the shell in real time. You can do this on a remote shell as well, but you might not be able to use your preferred code editor, and the script is deleted when the container is terminated.

File: docs/tasks/debug-application-cluster/local-debugging.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+You can exit the shell by typing `exit`.
```
Comment (radhikapc) [inline docs/tasks/debug-application-cluster/local-debugging.md]: Enter `exit` to quit and close the shell.

File: docs/tasks/debug-application-cluster/local-debugging.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## Developing or debugging an existing service
+
+When developing an application on Kubernetes, you typically code/debug a single service, but that service may require access to other services for proper testing and debugging. One option for coding and debugging is to use your continuous deployment pipeline, but even the fastest deployment pipeline introduces a delay in the code/debug cycle.
```
Comment (radhikapc) [inline docs/tasks/debug-application-cluster/local-debugging.md]: When developing an application on Kubernetes, you typically program or debug a single service. The service might require access to other services for testing and debugging. One option is to use the continuous deployment pipeline, but even the fastest deployment pipeline introduces a delay in the program or debug cycle.

File: docs/tasks/debug-application-cluster/local-debugging.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+With the `--swap-deployment` option, Telepresence lets you swap an existing deployment with the Telepresence proxy. This lets you run your service locally while connecting to the remote Kubernetes cluster -- while letting other services in the remote cluster also access your locally running instance. To run Telepresence with `--swap-deployment`, type:
```
Comment (radhikapc) [inline docs/tasks/debug-application-cluster/local-debugging.md]: Use the `--swap-deployment` option to swap an existing deployment with the Telepresence proxy. Swapping allows you to run a service locally and connect to the remote Kubernetes cluster. The services in the remote cluster can now access the locally running instance. 

To run `telepresence` with `--swap-deployment`, enter:

File: docs/tasks/debug-application-cluster/local-debugging.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+`telepresence --swap-deployment $DEPLOYMENT_NAME`
+
+where $DEPLOYMENT_NAME is the name of your existing deployment. This will spawn a shell. In the shell, start up your service. You can then make edits to your source code locally, save, and see the changes take effect immediately. You can also run your service in a debugger, or any other local development tool.
```
Comment (radhikapc) [inline docs/tasks/debug-application-cluster/local-debugging.md]: where $DEPLOYMENT_NAME is the name of your existing deployment. 

Running this command spawns a shell. In the shell, start your service. You can then make edits to the source code locally, save, and see the changes take effect immediately. You can also run your service in a debugger, or any other local development tool.

**Comment by richarddli** [comment]
OK, thank you. I've incorporated your feedback with minor edits.

**Comment by radhikapc** [comment]
@bowei please take a look at this PR

**Comment by bowei** [comment]
CC: @devin-donnelly 

I'm not sure this makes sense as part of the main k8s documentation as the tool does not ship with the main Kubernetes software.

**Comment by richarddli** [comment]
The section covers other non-native Kubernetes software (e.g., StackDriver, Elasticsearch, cAdvisor).

**Comment by richarddli** [comment]
Just wanted to bump to see if there was any feedback on this PR?

**Comment by bowei** [comment]
@devin-donnelly looping in the expert

**Comment by steveperry-53** [comment]
@richarddli @radhikapc @bowei, I'll take a stab at answering the question that was put to @devin-donnelly . In general, we are in the process of deciding where to put content that covers non-native Kubernetes software. But we haven't arrived at a firm policy yet. So I think it would be fine to merge this PR, and have this topic live in kubernetes.io. I think this topic is in line with other topics that live in the same section of the TOC. In the future, if we develop an overall strategy for separating core Kubernetes content from non-core Kubuenetes content, we can revisit the question of where this topic should live.

@richarddli Can you update your branch to fix the merge conflicts.

Then if I don't hear any objections, I'll go ahead and merge this.

**Comment by steveperry-53** [comment]
@richarddli I guess we still need approval from a tech reviewer. @bowei Can you do the tech review/approval? Thanks.

**Pull request merged**